File: 프로그래머스/Lv2/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int change(int n){
    int answer=1;
    
    while(n>1){
        if(n%2!=0) answer++;
        n=n/2;
    }
    
    return answer;
}

int solution(int n) {
    int answer = 0;
    int num = change(n);
    int temp=-1;
    int Index=0;
    
    while(num!=temp){
        Index++;
        temp = change(n+Index);
    }
    
    answer=n+Index;
    
    return answer;
}

File: 프로그래머스/Lv2/다음 큰 숫자.py
def solution(n):
    answer = 0
    arr=list(bin(n))
    
    for i in range(n+1,1000001):
        newArr=list(bin(i))
        if(arr.count('1')==newArr.count('1')):
            answer=int(''.join(newArr),2)
            break
    
    
    return answer
